rpc/src/error.rs

```diff
@@ -3,36 +3,110 @@ use ckb_tx_pool::error::Reject;
 use jsonrpc_core::{Error, ErrorCode, Value};
 use std::fmt::{Debug, Display};
 
-// * -1 ~ -999 General errors
-// * -1000 ~ -2999 Module specific errors
+/// CKB RPC error codes.
+///
+/// CKB RPC follows the JSON RPC specification about the [error object](https://www.jsonrpc.org/specification#error_object).
+///
+/// Besides the pre-defined errors, all CKB defined errors are listed here.
+///
+/// Here is a reference to the pre-defined errors:
+///
+/// | code             | message          | meaning                                            |
+/// | ---------------- | ---------------- | -------------------------------------------------- |
+/// | -32700           | Parse error      | Invalid JSON was received by the server.           |
+/// | -32600           | Invalid Request  | The JSON sent is not a valid Request object.       |
+/// | -32601           | Method not found | The method does not exist / is not available.      |
+/// | -32602           | Invalid params   | Invalid method parameter(s).                       |
+/// | -32603           | Internal error   | Internal JSON-RPC error.                           |
+/// | -32000 to -32099 | Server error     | Reserved for implementation-defined server-errors. |
+///
+/// CKB application defined errors follow some patterns to assign the codes:
+///
+/// * -1 ~ -999 are general errors
+/// * -1000 ~ -2999 are module specific errors. Each module generally gets 100 reserved error
+/// codes.
+///
+/// Unless otherwise noted, all the errors returns optional detailed information as `string` in the error
+/// object `data` field.
 #[derive(Debug, PartialEq, Clone, Copy)]
 pub enum RPCError {
-    // ,-- General application errors
+    /// (-1): CKB internal errors which are considered to never happen or only happen when the system
```

Comment:
The RPC doc does not show the enum integer value, so I have to duplicate it in the comment.